pkg/util/tools/bijection.go

```diff
@@ -0,0 +1,93 @@
+package tools
```

Comment:
Just make it `pkg/util/bijection`? Otherwise you are basically creating another utils inside utils, which is no better than putting them in `util` directly.